Setup an actual build system

still debugging

```
$ npm run export -- --mulOnly
> phetscraper@1.0.0 export /home/travis/build/openzim/phet
> npm run export-prebuild && node --no-warnings --max-old-space-size=8192 node_modules/ts-node/dist/bin.js steps/export.ts "--mulOnly"
> phetscraper@1.0.0 export-prebuild /home/travis/build/openzim/phet
> npm run build && node_modules/browserify/bin/cmd.js res/js/index.js -t babelify --presets [es2015] -o res/js/dist.js
> phetscraper@1.0.0 build /home/travis/build/openzim/phet
> tsc
/home/travis/build/openzim/phet/node_modules/ts-node/src/index.ts:556
          throw new TypeError(`${relative(cwd, fileName)}: Emit skipped`)
                ^
TypeError: lib/logger.js: Emit skipped
    at getOutput (/home/travis/build/openzim/phet/node_modules/ts-node/src/index.ts:556:17)
    at Object.compile (/home/travis/build/openzim/phet/node_modules/ts-node/src/index.ts:758:32)
    at Module.m._compile (/home/travis/build/openzim/phet/node_modules/ts-node/src/index.ts:837:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Object.require.extensions.(anonymous function) [as .js] (/home/travis/build/openzim/phet/node_modules/ts-node/src/index.ts:840:12)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! phetscraper@1.0.0 export: `npm run export-prebuild && node --no-warnings --max-old-space-size=8192 node_modules/ts-node/dist/bin.js steps/export.ts "--mulOnly"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the phetscraper@1.0.0 export script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-04-03T00_33_54_205Z-debug.log
Done. Your build exited with 0.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,4 +2,4 @@ language: node_js
 node_js:
   - "10"
 before_script: "npm run setup"
-script: "npm run get -- --includeLanguages en && npm run transform && npm run export -- --includeLanguages en --mulOnly"
+script: "npm run get -- --includeLanguages en && npm run transform && npm run export -- --mulOnly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,10 @@
 language: node_js
 node_js:
-  - "6"
-  - "7"
-script: "npm run build"
+  - "10"
+env:
+  - "PHET_RPS=4"
+  - "PHET_WORKERS=2"
+  - "PHET_RETRIES=5"
+  - "PHET_VERBOSE_ERRORS=true"
+before_script: "npm run setup"
+script: "npm run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,4 +2,4 @@ language: node_js
 node_js:
   - "10"
 before_script: "npm run setup"
-script: "npm run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"
+script: "npm run get -- --includeLanguages en && npm run transform && npm run export -- --includeLanguages en --mulOnly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,4 +2,4 @@ language: node_js
 node_js:
   - "10"
 before_script: "npm run setup"
-script: "export PHET_VERBOSE_ERRORS=true npm run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"
+script: "npm run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,4 +2,5 @@ language: node_js
 node_js:
   - "10"
 before_script: "npm run setup"
-script: "npm run get -- --includeLanguages en && npm run transform && npm run export -- --mulOnly"
+script: "npm run get -- --includeLanguages en && npm run transform"
+after_script: "npm run export -- --mulOnly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,10 +1,5 @@
 language: node_js
 node_js:
   - "10"
-env:
-  - "PHET_RPS=4"
-  - "PHET_WORKERS=2"
-  - "PHET_RETRIES=5"
-  - "PHET_VERBOSE_ERRORS=true"
 before_script: "npm run setup"
-script: "npm run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"
+script: "export PHET_VERBOSE_ERRORS=true run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,4 +2,4 @@ language: node_js
 node_js:
   - "10"
 before_script: "npm run setup"
-script: "export PHET_VERBOSE_ERRORS=true run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"
+script: "export PHET_VERBOSE_ERRORS=true npm run get -- --includeLanguage en && npm run transform && npm run export -- --includeLanguage en --mulOnly"

--- a/lib/welcome.ts
+++ b/lib/welcome.ts
@@ -1,5 +1,5 @@
-import * as v8 from "v8"
-import {log} from "./logger"
+import * as v8 from 'v8';
+import {log} from './logger';
 
 export default (stage) => {
     log.info(`Node ${process.version}`);

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -5,7 +5,7 @@
         "sourceMap": false,
         "noImplicitAny": false,
         "suppressImplicitAnyIndexErrors": true,
-        "allowJs": true,
+        "allowJs": false,
         "module": "commonjs",
         "target": "ES5",
         "skipLibCheck": true,

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -12,6 +12,9 @@
         "resolveJsonModule": true
     },
     "files": [
-        "res/js/index.ts"
+        "res/js/index.ts",
+        "steps/get.ts",
+        "steps/transform.ts",
+        "steps/export.ts"
     ]
 }

